R/prep_data.R

```diff
@@ -264,6 +266,11 @@ prep_data <- function(d,
 
     # If there are nominal predictors, apply nominal transformations
     if (any(var_info$type == "nominal" & var_info$role == "predictor")) {
+
+      # Add protective levels --------------------------------------------------
```

Comment:
These kinds of comment headers are crufty and are code smells that refactoring needs to take place. We should consider breaking this 225 line function down and let the smaller functions *be the comment*.

Reply:
That should happen in another PR that I'm happy to submit after this is merged.

Reply:
I'm all for it.